src/sqladmin-fetcher.ts

```diff
@@ -63,6 +85,8 @@ export class SQLAdminFetcher {
     regionId,
     instanceId,
   }: InstanceConnectionInfo): Promise<InstanceMetadata> {
+    setupGaxiosConfig();
```

Comment:
should we look at maybe having this setup as part of the `SQLAdminFetcher` constructor? That way we can call `setupGaxiosConfig()` and `cleanGaxiosConfig()` only once instead of in each method (`getEphemeralCertificate`, and `getInstanceMetadata`)?

Reply:
Yeah, I tried to bring up the reason for doing a setup/cleanup per method in the comment at L52 (in case you want to read that and let me know if it's clear enough / makes sense) 😊 

The long explanation is that `gaxios` _optionally_ share a single instance (well, as long as the library is not duplicated in `node_modules` which _may_ happen and also as long as the user is not creating a new Gaxios instance instead of using the exposed `request` method, for ref most of our libraries use the public `request` method so they use this shared instance) for its requests and with its current provided API (`instance.defaults` that I'm using here) this customize config may be applied to any concurrent requests.

By having the setup as the first step of each method and cleaning it up once the method is done I'm trying to avoid as much as possible having this custom config be applied to other different requests that might be happening either elsewhere in our library or in the user code - keep in mind that the user code may also be using this shared `gaxios` instance.

Reply:
Gotcha that makes sense, since the SQLAdminFetcher is long living you want to return it back to the defaults between refreshes and only update its config during a refresh basically.